### PR TITLE
Retry ValidatorStatusLogger initial status check until beacon node is ready

### PR DESCRIPTION
**Pull request by cemozerr**
## PR Description
Retry ValidatorStatusLogger initial status check until the beacon node is ready.

## Documentation

- [x] I thought about documentation and added the `documentation` label to this PR if updates are required.

### CODE REVIEW

File: validator/client/src/main/java/tech/pegasys/teku/validator/client/DefaultValidatorStatusLogger.java [hunk 1]
```diff
@@ -19,48 +19,66 @@
...
   }
 
   @Override
-  public void printInitialValidatorStatuses() {
-    validatorApiChannel
+  public SafeFuture<Void> printInitialValidatorStatuses() {
+    return validatorApiChannel
         .getValidatorStatuses(validatorPublicKeys)
-        .thenAccept(
+        .thenCompose(
```
Comment (mbaxter) [inline validator/client/src/main/java/tech/pegasys/teku/validator/client/DefaultValidatorStatusLogger.java:59]: I recently added a retry utility (`AsyncRunner.runWithRetry`) that might be useful here.

**Review by mbaxter — APPROVED** [review]

File: validator/coordinator/src/main/java/tech/pegasys/teku/validator/coordinator/ValidatorApiHandler.java [hunk 2]
```diff
@@ -216,20 +216,22 @@ public ValidatorApiHandler(
...
-                                toMap(
-                                    ValidatorResponse::getPublicKey,
-                                    ValidatorResponse::getStatus))));
+    return isSyncActive()
+        ? SafeFuture.completedFuture(Optional.empty())
+        : chainDataProvider
+            .getStateValidators(
+                "head",
+                validatorIdentifiers.stream().map(BLSPublicKey::toString).collect(toList()),
+                new HashSet<>())
```
Comment (ajsutton) [inline validator/coordinator/src/main/java/tech/pegasys/teku/validator/coordinator/ValidatorApiHandler.java:225]: I'm very unclear how we got a `ChainDataProvider` in here - that should be part of the REST API implementation and the dependency should go the other way around where the providers call into `ValidatorApiChannel`.  We shouldn't be using the REST API in the internals of Teku like this (ideally that `ValidatorResponse` class shouldn't be accessible at all from here).  Actually same with `ValidatorStatus`, it shouldn't be used from here either.

Probably need to rework this in a follow up PR and get the dependency chain right.

File: validator/remote/src/main/java/tech/pegasys/teku/validator/remote/RemoteValidatorApiHandler.java [hunk 3]
```diff
@@ -100,25 +100,34 @@ public RemoteValidatorApiHandler(
...
           publicKeys.subList(i, Math.min(publicKeys.size(), i + MAX_PUBLIC_KEY_BATCH_SIZE));
-      requestValidatorObject(batch, valueExtractor).ifPresent(returnedObjects::putAll);
+      Optional<Map<BLSPublicKey, T>> validatorObjects =
+          requestValidatorObject(batch, valueExtractor);
+      boolean requestSuccesful = validatorObjects.isPresent();
+      if (requestSuccesful) {
+        returnedObjects.putAll(validatorObjects.get());
+      } else {
+        return Optional.empty();
+      }
```
Comment (ajsutton) [inline validator/remote/src/main/java/tech/pegasys/teku/validator/remote/RemoteValidatorApiHandler.java]: nit:
```suggestion
      if (validatorObjects.isEmpty()) {
        return Optional.empty();
      }
      returnedObjects.putAll(validatorObjects.get());
```

**Review by ajsutton — APPROVED** [review]
LGTM. Meredith's suggestion about `runWithRetry` is definitely worth looking into as well.

File: validator/remote/src/main/java/tech/pegasys/teku/validator/remote/RemoteValidatorApiHandler.java (hunk 3 above)
Comment (cemozerr) [reply validator/remote/src/main/java/tech/pegasys/teku/validator/remote/RemoteValidatorApiHandler.java]: Done.

File: validator/coordinator/src/main/java/tech/pegasys/teku/validator/coordinator/ValidatorApiHandler.java (hunk 2 above)
Comment (cemozerr) [reply validator/coordinator/src/main/java/tech/pegasys/teku/validator/coordinator/ValidatorApiHandler.java:225]: I believe I had inserted the ChainDataProvider to ValidatorApiHandler (we create a new one using combined chain data client and recent chain data client in BeaconChainController). The reason was that using ChainDataProvider methods made it easy to draw parallels when implementing the ValidatorApiChannel interface methods for the RemoteValidatorApiHandler and ValidatorApiHandler.

Comment (cemozerr) [reply validator/coordinator/src/main/java/tech/pegasys/teku/validator/coordinator/ValidatorApiHandler.java:225]: I understand your point though and can revert that in another PR.

**Pull request merged**